pkgs/by-name/sc/scopehal-apps/package.nix

```diff
@@ -78,10 +84,31 @@ stdenv.mkDerivation {
     moltenvk
   ];
 
-  cmakeFlags = lib.optionals stdenv.hostPlatform.isDarwin [
-    "-DCMAKE_INSTALL_RPATH=${lib.strings.makeLibraryPath [ vulkan-loader ]}"
+  cmakeFlags = [
+    "-DNGSCOPECLIENT_VERSION=${version}"
+  ];
+
+  patches = [
+    ./remove-git-derived-version.patch
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    ./remove-brew-molten-vk-lookup.patch
   ];
 
+  postFixup = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    mv -v $out/bin/ngscopeclient $out/bin/.ngscopeclient-unwrapped
+    makeWrapper $out/bin/.ngscopeclient-unwrapped $out/bin/ngscopeclient \
+      --prefix DYLD_LIBRARY_PATH : "${lib.makeLibraryPath ([ vulkan-loader ])}"
+  '';
+
+  postInstall = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    mkdir -p $out/Applications/ngscopeclient.app/Contents/{MacOS,Resources}
+
+    install -m644 {../src/ngscopeclient/icons/macos,$out/Applications/ngscopeclient.app/Contents/Resources}/ngscopeclient.icns
+
+    write-darwin-bundle $out ngscopeclient ngscopeclient ngscopeclient
```

Comment:
`write-darwin-bundle` will build its own Info.plist, instead of using the project-supplied one, but I think that's ok.